zcash_client_backend/src/scanning.rs

```diff
@@ -120,7 +120,7 @@ pub struct ScanningKey<Ivk, Nk, AccountId> {
     key_scope: Option<Scope>,
 }
 
-impl<AccountId> ScanningKeyOps<SaplingDomain, AccountId, sapling::Nullifier>
+impl<AccountId: Send + Sync> ScanningKeyOps<SaplingDomain, AccountId, sapling::Nullifier>
```

Comment:
Is this bound (repeated everywhere) avoidable if we add the equivalent bound on `AccountId` in `ScanningKeyOps`?

Reply:
Unfortunately, to be able to move an instance of `ScannningKeys` to `tokio::task::spawn_blocking` for example, the compiler requires `Send` and `Sync` to be implemented for `ScanningKeyOps`. Otherwise, I get this error:

```md
`(dyn ScanningKeyOps<sapling_crypto::note_encryption::SaplingDomain, AccountId, sapling_crypto::note::nullifier::Nullifier> + 'static)` cannot be sent between threads safely
```

I tried bounding `AccountId` by `Send` and `Sync` in `ScanningKeyOps`, but that didn't help.

Bounding `ScanningKeyOps` itself eliminates the error, but once I do that, I have to make the rest of the changes in this PR to fix subsequent bound requirements.

Reply:
This PR is not important though, feel free to close it without merging. The workaround was to create an instance of `ScanningKeys` in the scanning thread itself. Moreover, we have no plans to work on the scanner any time soon.